src/lsm/compaction.zig

```diff
@@ -315,30 +334,49 @@ pub fn CompactionType(
             assert(compaction.io_pending == 0);
             assert(!compaction.merge_done);
 
-            compaction.callback = callback;
-
-            tracer.start(
-                &compaction.tracer_slot,
-                .{ .tree_compaction = .{ .compaction_name = compaction.name } },
-                .{ .tree_compaction_tick = .{ .level_b = compaction.level_b } },
-                @src(),
-            );
-
-            // Generate fake IO to make sure io_pending doesn't reach zero multiple times from
-            // IO being completed inline down below.
-            // The fake IO is immediately resolved and triggers the cpu_merge_start if all
-            // IO completes inline or if no IO was started.
-            compaction.io_start();
-            defer compaction.io_finish();
-
-            // Start reading blocks from the iterators to merge them.
-            if (compaction.iterator_a.tick()) compaction.io_start();
-            if (compaction.iterator_b.tick()) compaction.io_start();
-
-            // Start writing blocks prepared by the merge iterator from a previous compact_tick().
-            compaction.io_write_start(.data);
-            compaction.io_write_start(.filter);
-            compaction.io_write_start(.index);
+            switch (compaction.strategy) {
+                .move => {
+                    const snapshot_max = snapshot_max_for_table_input(compaction.op_min);
+                    const level_b = compaction.level_b;
+                    const level_a = level_b - 1;
+
+                    const table_a = compaction.level_a_input.?;
```

Comment:
Looks like this is copying the `TableInfo` to the stack? Can't we take a constant pointer here instead?